### PR TITLE
feat: search websites by name, url, etc.

### PR DESCRIPTION
**Pull request by alan910127**
---

<details open="true"><summary>Generated summary (powered by <a href="https://app.graphite.dev">Graphite</a>)</summary>

> ## TL;DR
> This pull request includes changes to the "routes.py" file in a Python application. The changes involve importing modules, modifying imports, and defining routes and functions for a website API. It also includes changes to the Pagination class in the schemas.py file and the addition of a new file called "__init__.py" in the "search" directory.
> 
> ## What changed
> - Imported modules and classes such as SearchResultWebsite, UpdateResponse, UpdateWebsitePayload, and SearchWebsiteCursor.
> - Defined a router for an API endpoint related to website updates.
> - Defined a ParseCursorError class and a SearchWebsiteCursor class, along with their methods.
> - Modified the Pagination class in the schemas.py file.
> - Added a new file called "__init__.py" to the "search" directory.
> 
> ## How to test
> 1. Ensure all the required modules and classes are imported correctly.
> 2. Test the defined routes and functions for the website API.
> 3. Verify that the ParseCursorError and SearchWebsiteCursor classes and their methods are working as expected.
> 4. Check if the Pagination class in the schemas.py file has been modified correctly.
> 5. Test the functionality of the "__init__.py" file in the "search" directory.
> 
> ## Why make this change
> - The changes were made to improve the functionality and organization of the code.
> - Importing the necessary modules and classes ensures that the required dependencies are available.
> - Defining routes and functions for the website API allows for handling specific requests and performing necessary operations.
> - The addition of the ParseCursorError and SearchWebsiteCursor classes provides error handling and cursor parsing functionality.
> - Modifying the Pagination class in the schemas.py file improves the naming consistency and clarity of the code.
> - The addition of the "__init__.py" file in the "search" directory allows for easier importing of the "search_websites" function from the "db" module.
</details>

### CODE REVIEW

**Comment by alan910127** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #1** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/1?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/3?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #4** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/4?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #5** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/5?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #6** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/6?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/GymSquad/archive-api/6?utm_source=stack-comment).

File: app/websites/routes.py [hunk 1]
```diff
@@ -133,31 +69,65 @@ async def search_websites(
...
+        result=result,
         pagination=Pagination(
-            next_cursor="",
-            has_next=False,
-            num_results=0,
-            total_results=0,
+            next_cursor=searched.next_cursor.to_base64()
+            if searched.next_cursor is not None
+            else None,
+            has_next=searched.next_cursor is not None,
```
Comment (alan910127) [inline app/websites/routes.py]: `has_next` cannot be extracted, since the type system cannot infer`next_cursor is not None` from `has_next == True`

**Review by BaBa0525 — APPROVED** [review]
[ci](https://app.graphite.dev/macros/GymSquad)

**Comment by alan910127** [comment]
### Merge activity

* **Jan 27, 6:16 AM**: @alan910127 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/GymSquad/archive-api/6).
* **Jan 27, 6:23 AM**: [Graphite](https://app.graphite.dev/github/pr/GymSquad/archive-api/6) rebased this pull request as part of a merge.
* **Jan 27, 6:24 AM**: @alan910127 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/GymSquad/archive-api/6).

**Pull request merged**